scheds/rust/scx_rusty/src/main.rs

```diff
@@ -563,46 +604,263 @@ impl<'a, 'b, 'c> LoadBalancer<'a, 'b, 'c> {
         }
     }
 
+    fn bucket_range(&self, bucket: u64) -> (f64, f64) {
+        const MAX_WEIGHT: u64 = bpf_intf::consts_LB_MAX_WEIGHT as u64;
+        const NUM_BUCKETS: u64 = bpf_intf::consts_LB_LOAD_BUCKETS as u64;
+        const WEIGHT_PER_BUCKET: u64 = MAX_WEIGHT / NUM_BUCKETS;
+
+        if bucket >= NUM_BUCKETS {
+            panic!("Invalid bucket {}, max {}", bucket, NUM_BUCKETS);
+        }
+
+        // w_x = [1 + (10000 * x) / N, 10000 * (x + 1) / N]
+        let min_w = 1 + (MAX_WEIGHT * bucket) / NUM_BUCKETS;
+        let max_w = min_w + WEIGHT_PER_BUCKET - 1;
+
+	(min_w as f64, max_w as f64)
+    }
+
+    fn bucket_weight(&self, bucket: u64) -> f64 {
+        const WEIGHT_PER_BUCKET: f64 = bpf_intf::consts_LB_WEIGHT_PER_BUCKET as f64;
+        let (min_weight, _) = self.bucket_range(bucket);
+
+	// Use the mid-point of the bucket when determining weight
+        min_weight + (WEIGHT_PER_BUCKET / 2.0f64)
+    }
+
+    fn apply_infeas_threshold(&mut self,
+                             doms_dcycles_buckets: &[f64],
+                             infeas_thrsh: &f64) {
```

Comment:
What's with taking a single value type as a reference and then cloning it? Is there some advantage for doing that?

Reply:
Sorry, this is left over from a revision where I'd fall back to the caller. Thanks for noticing, will update.